app/js/FSPersistor.js

```diff
@@ -80,6 +92,38 @@ async function getFileSize(location, filename) {
   }
 }
 
+async function getFileMd5Hash(location, filename) {
+  const fullPath = path.join(location, filterName(filename))
+  try {
+    return await _getFileMd5HashForPath(fullPath)
+  } catch (err) {
+    throw new ReadError({
+      message: 'unable to get md5 hash from file',
+      info: { location, filename }
+    }).withCause(err)
+  }
+}
+
+async function _getFileMd5HashForPath(fullPath) {
+  return new Promise((resolve, reject) => {
+    try {
+      const readStream = fs.createReadStream(fullPath)
+      const hash = crypto.createHash('md5')
+      hash.setEncoding('hex')
+      readStream.on('end', () => {
+        hash.end()
+        resolve(hash.read())
+      })
+      readStream.on('error', err => {
+        throw err
+      })
+      readStream.pipe(hash)
```

Comment:
In order to catch the `err` of the `readStream.on('error'...)` handler in the `catch` block below, we need to use `Stream.pipeline` here, no?

Reply:
Yeah, the format of this method is a bit weird. There's not really a need to try/catch at all - can just use reject on error if pipeline fails.